pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -313,13 +315,38 @@ def store_predictions(self, output, batch_idx, dataloader_idx):
 
     def on_evaluation_epoch_end(self, *args, **kwargs):
         # call the callback hook
-        if self.trainer.testing:
-            self.trainer.call_hook('on_test_epoch_end', *args, **kwargs)
-        else:
-            self.trainer.call_hook('on_validation_epoch_end', *args, **kwargs)
+        self.call_on_evaluation_epoch_end_hook()
 
         self.trainer.call_hook('on_epoch_end')
 
+    def call_on_evaluation_epoch_end_hook(self):
+        outputs = self.outputs
+        model_ref = self.trainer.get_model()
+        hook_name = "on_test_epoch_end" if self.trainer.testing else "on_validation_epoch_end"
+
+        self.trainer._reset_result_and_set_hook_fx_name(hook_name)
+
+        with self.trainer.profiler.profile(hook_name):
+
+            if hasattr(self.trainer, hook_name):
+                on_evaluation_epoch_end_hook = getattr(self.trainer, hook_name)
+                on_evaluation_epoch_end_hook(outputs)
+
+            if is_overridden(hook_name, model_ref):
+                model_hook_fx = getattr(model_ref, hook_name)
+                model_hook_params = list(inspect.signature(model_hook_fx).parameters)
+                if "args" in model_hook_params or "outputs" in model_hook_params:
+                    model_hook_fx(outputs)
```

Comment:
I think you can make the method `is_param_in_hook_signature` a normal function and use it again here. That way it can be used while deprecating other signatures too.

Reply:
Moving it to utilities sounds good?

Reply:
yeap

Reply:
Added

Reply:
@rohitgr7 Added